### ISSUE TITLE
Add `--verbose` flag to lint, showing copyright/license info sources

### ISSUE THREAD

**Issue by robinkrahl** [fsfe/reuse-tool#256]
If a project uses multiple licenses, it is hard to get an overview of which files are under which license without manually inspecting the files.  `reuse lint` only lists all used licenses, and while `reuse spdx` lists the license information for all files, it is rather hard to read for a human.

I think it would be good to have a `reuse list`command that prints a table with the copyright information for all files (or maybe `report` to increase the edit distance from `lint`) :

```
$ reuse list
File	License		Copyright Text
1.c	MIT		2020 John Doe
2.c	MIT		2020 John Doe
3.txt	CC0-1.0		2019 Jane Smith
```

It would be even better if the command could group files with the same copyright information:

```
$ reuse list
File	License		Copyright Text
*.c	MIT		2020 John Doe
3.txt	CC0-1.0		2019 Jane Smith
```

Use cases:
1. Getting an overview of a third-party project.
2. Making sure that you added the correct annotations for your own project.

I can try to implement this if you would merge it.

**Comment by mxmehl** [comment]
Indeed, that'd be a nice view, also for debugging.

Perhaps related to #183 since `reuse spdx` should already include all information. Outputting it as JSON (and/or YAML) would allow for a simpler manipulation that could also be done manually, e.g. with tools like `jq`.

**Comment by robinkrahl** [comment]
Yes, using a separate parser for the `reuse spdx` output would also be a way to implement this.  Does this mean you’d rather not have this feature in reuse-tool directly?

**Comment by mxmehl** [comment]
> Does this mean you’d rather not have this feature in reuse-tool directly?

I am fine with having this function, either as a full list (as per your first post), or also with the option to have a detailed view for a single file (e.g. explaining the chain of precedence in case there are multiple explicit/glob coverages for this file).

Regarding spdx, I just wanted to make sure that we do not forget that users also may have this function to build their own custom query if necessary.

**Comment by mxmehl** [comment]
In today' hackathon we concluded that we would like to have this verbose listing as part of the `lint` subcommand, e.g. `reuse lint --verbose`, perhaps also combinable with `--json`.

Aside from the detected license and copyright it should also provide info on *where* this information has been derived from, e.g. the file itself, the .license file, or dep5 (or future REUSE.yaml).

**Comment by mxmehl** [comment]
So us maintainers would like to see this feature in the next release.

What's unclear yet:
* Should this be a separate subcommand (e.g. `list`) or a flag to `lint` like `--verbose`? We had a preference for the flag in the January call but in the light of recent developments this may have changed.
* Can we make it work together with #512 and #183?

What we agreed on:
* This should not be the default output for `lint` if it would become a flag for this subcommand. Also not when just linting a single file, like proposed in #512

**Comment by mxmehl** [comment]
In today's call we clarified a few more questions:
* The desired functionality shall happen via the `--verbose` flag to the `lint` subcommand, not to a separate subcommand.
* It shall happen in the same step as #183 as the `--json` flag shall imply `--verbose` already.

So this issue can be used to track the effort of adding the verbose output to the *plain*, human-readable output of `reuse lint`.
